packages/bytecode-utils/test/bytecode.spec.ts

```diff
@@ -42,39 +60,84 @@ describe('bytecode utils', function () {
 
   it('bytecode decode cbor with `ipfs` property', () => {
     chai
-      .expect(decode(BYTECODE_IPFS).ipfs)
+      .expect(decode(BYTECODE_IPFS, AuxdataStyle.SOLIDITY).ipfs)
       .to.equal('QmdD3hpMj6mEFVy9DP4QqjHaoeYbhKsYvApX1YZNfjTVWp');
   });
 
   it('bytecode decode cbor with `bzzr1` property', () => {
     chai
-      .expect(decode(BYTECODE_BZZR1).bzzr1)
+      .expect(decode(BYTECODE_BZZR1, AuxdataStyle.SOLIDITY).bzzr1)
       .to.equal(
         '0x71e0c183217ae3e9a1406ae7b58c2f36e09f2b16b10e19d46ceb821f3ee6abad',
       );
   });
 
   it('bytecode decode cbor with `experimental` property', () => {
-    chai.expect(decode(BYTECODE_EXPERIMENTAL).experimental).to.be.true;
+    chai.expect(
+      decode(BYTECODE_EXPERIMENTAL, AuxdataStyle.SOLIDITY).experimental,
+    ).to.be.true;
+  });
+
+  it('bytecode decode Vyper cbor auxdata for version >= 0.4.1', () => {
+    chai
+      .expect(decode(BYTECODE_VYPER_INTEGRITY, AuxdataStyle.VYPER))
+      .to.deep.equal({
+        integrity: new Uint8Array([
+          5, 183, 84, 197, 139, 46, 84, 10, 20, 171, 166, 241, 103, 23, 171, 44,
+          48, 237, 199, 73, 54, 200, 152, 93, 119, 177, 82, 205, 151, 136, 126,
+          7,
+        ]),
+        runtimeSize: 143,
+        dataSizes: [],
+        immutableSize: 0,
+        compiler: '0.4.1',
+      });
+  });
+
+  it('bytecode decode Vyper cbor auxdata for version >= 0.3.10 and < 0.4.1', () => {
+    chai
+      .expect(decode(BYTECODE_VYPER_NO_INTEGRITY, AuxdataStyle.VYPER))
+      .to.deep.equal({
+        runtimeSize: 143,
+        dataSizes: [],
+        immutableSize: 0,
+        compiler: '0.3.10',
+      });
+  });
+
+  it('bytecode decode Vyper cbor auxdata for version < 0.3.10', () => {
+    chai
+      .expect(decode(BYTECODE_VYPER_NO_ARRAY, AuxdataStyle.VYPER))
+      .to.deep.equal({
+        compiler: '0.3.8',
+      });
+  });
+
+  it('bytecode decode Vyper cbor auxdata for version < 0.3.5', () => {
+    chai
+      .expect(decode(BYTECODE_VYPER_NO_AUXDATA_LENGTH, AuxdataStyle.VYPER))
+      .to.deep.equal({
+        compiler: '0.3.4',
+      });
   });
```

Comment:
Because for Vyper version less than 0.3.5 there is no auxdata length, even if there is the auxdata itself

```
gte(0.3.5)
<bytecode><auxadata><two_bytes_auxdata_length>

lt(0.3.5)
<bytecode><auxadata>